### PR TITLE
Feat: Associations

### PR DESCRIPTION
**Pull request by leehaney254**
- Created a class Classroom 
- Created the has-many/belongs-to relationship between Classroom and Student
- Created a class Book
- Created a class Rental
- Create the many-to-many relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentnals = []
+  end
+
+  def add_rented(rental)
+    @rentnals << rental
+  end
+end
```
Comment (DammyShittu) [inline book.rb]: - [ ] Kindly add the other end of the has many relationships by adding the `rent` method as you have in the person class.

  ```
  def rent(person, date)
    Rental.new(person, self, date)
  end
  ```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  attr_accessor :title, :author, :rentnals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentnals = []
+  end
+
+  def add_rented(rental)
+    @rentnals << rental
```
Comment (DammyShittu) [inline book.rb]: - **[OPTIONAL]** I think this should be `rentals`, not `rentnals`

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @leehaney254,

## Highlights
✔️ Creating associations

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by fobadara — APPROVED** [review]
Hi @leehaney254, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Linters are passing
✅ Implement reviewer's suggestion properly

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**